irohad/main/irohad.cpp

```diff
@@ -159,6 +172,62 @@ static shared_model::crypto::Keypair getKeypairFromFile(
              });
 }
 
+boost::optional<std::string> getEnvVar(std::string const& key)
+{
+    char const* val = getenv(key.c_str()); 
+    return val == NULL ? boost::optional<std::string>() : std::string(val);
+}
+
+boost::optional<IrohadConfig::DbConfig> getPostgresCredsFromEnv() {
+  
+  auto pg_host = getEnvVar("IROHA_POSTGRES_HOST");
+  auto pg_port_str = getEnvVar("IROHA_POSTGRES_PORT");
+  auto pg_user = getEnvVar("IROHA_POSTGRES_USER");
+  auto pg_pass = getEnvVar("IROHA_POSTGRES_PASSWORD");
+  auto pg_w_dbname = getEnvVar("IROHA_POSTGRES_DATABASE");
+  auto pg_m_dbname = getEnvVar("IROHA_POSTGRES_MAINTENANCE_DATABASE");
+
+  std::uint16_t pg_port;
+  if (pg_port_str.has_value()) {
+    auto pg_port_int = std::stoi(pg_port_str.value());
+    if (pg_port_int < std::numeric_limits<std::uint16_t>::min() ||
+        pg_port_int > std::numeric_limits<std::uint16_t>::max()) {
+      std::ostringstream os;
+      os << "The IROHA_POSTGRES_PORT environment variable value (" << pg_port_int
+          << ") is out of range.";
+      throw std::invalid_argument(std::move(os).str());
```

Comment:
I'd suggest not to create heavy `ostringstream` but to use: [std::to_string](https://en.cppreference.com/w/cpp/string/basic_string/to_string).
```suggestion
      std::ostringstream os;
      os << "The IROHA_POSTGRES_PORT environment variable value (" << pg_port_int
          << ") is out of range.";
     auto errorMessage = "The IROHA_POSTGRES_PORT environment variable value (" + std::to_string(pg_port_int) + ") is out of range.";
      throw std::invalid_argument(std::move(errorMessage));
```

Reply:
I'd suggest to output `pg_port_str` to error message. And what "range" do U mean. Please tell that PORT MIUST be from 0 to 65535.

Reply:
there is also a little hack to check that port is in range 
`pg_port_int == static_cast<std::uint16_t>(pg_port_int);`

Reply:
@baziorek @kuvaldini Thank you both for the improvements, done: ff2643ea64c42690445dc84d48c606835a79ce88